### PR TITLE
Add codacy support

### LINKED ISSUE #10: Setup codacy
### Description

In order to make sure all code is written in good quality, codacy needs to be used so that each PR is validated.

### PR DESCRIPTION
**Pull request by m-linner-ericsson**
### Applicable Issues
This will fix #10 

### Description of the Change
In Codacy I have setup a link to this repo. For more information how this was done please see https://github.com/eiffel-community/community/pull/21

Things changed in files:
- Excluded the Github templates and CODE_OF_CONDUCT as they will interfere with codacy style checks. I didn't want to change the names and didn't have time to look into the issue deeper
- Changed the travis file to include running of coverage and uploading it to Codacy
- Fixed issues found by the markdown linter

### Alternate Designs
None

### Benefits
Style check on each pull request and code coverage will ensure that the project will not deviate.

### Possible Drawbacks
None

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Mattias Linnér mattias.linner@ericsson.com

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,42 +24,42 @@ The following is a set of guidelines for contributing to this repository.
 Anyone is welcome to propose changes to the contents of this repository by creating a new [Issue](https://github.com/eiffel-community/eiffel-gerrit-plugin/issues) ticket in GitHub. These requests may concern anything contained in the repo: changes to documentation, bug fixes, requests for additional information, additional event types, requests for additional examples, new featuers et cetera.
 
 When posting a new issue, try to be as precise as possible and phrase your arguments for your request carefully. Keep in mind that collaborating on software development is often an exercise in finding workable compromises between multiple and often conflicting needs; this is particularly true for defining a shared protocol such as Eiffel. In particular, pay attention to the following:
-1. What type of change is requested?
-1. Why would you like to see this change?
-1. Can you provide any concrete examples?
-1. Which arguments in favor can you think of?
-1. Which arguments against can you think of, and why are they outweighed by the arguments in favor?
+1.  What type of change is requested?
```
Comment (e-pettersson-ericsson) [inline CONTRIBUTING.md]: How about making a PR to the community including these changes. Should it be in the template repository I guess? So we can cascade these changes to all repositories in the eiffel community?

Comment (SantoshNC68) [reply CONTRIBUTING.md]: Sure great idea, we can raise an issue in community and can also raise a PR there to fix.

Comment (m-linner-ericsson) [reply CONTRIBUTING.md]: I will do that

Comment (m-linner-ericsson) [reply CONTRIBUTING.md]: https://github.com/eiffel-community/eiffel-repository-template/pull/3

**Pull request merged**